lib/core/src/Cardano/Wallet/DB/MVar.hs

```diff
@@ -121,8 +125,10 @@ newDBLayer = do
                         Right $ Just $ Database cp meta (txs' <> txs) k
             txs' `deepseq` alterMVar db alter key
 
-        , readTxHistory = \key ->
-            maybe mempty txHistory . Map.lookup key <$> readMVar db
+        , readTxHistory = \key -> let
+                sortedTxHistory = sortOn slot . Map.toList . txHistory
+                slot = Down . slotId . snd . snd
+            in maybe mempty sortedTxHistory . Map.lookup key <$> readMVar db
```

Comment:
Side-note: we should probably get rid of the DB MVar implementation at some point.